src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -17,14 +17,20 @@
 /**
  * InputBag is a container for user input values such as $_GET, $_POST, $_REQUEST, and $_COOKIE.
  *
+ * @template TInput of string|int|float|bool|null
+ *
  * @author Saif Eddin Gmati <azjezz@protonmail.com>
  */
 final class InputBag extends ParameterBag
 {
     /**
      * Returns a scalar input value by name.
      *
-     * @param string|int|float|bool|null $default The default value if the input key does not exist
+     * @template TDefault of TInput|null
+     *
+     * @param TDefault $default The default value if the input key does not exist
+     *
+     * @return TDefault|TInput
```

Comment:
I don’t get those types: `TInput` already includes `null`, so aren’t `TDefault` and `TDefault|TInput` the same?

Reply:
Let's take a `InputBag<string>`
this method can return either `string`, or the type of `$default`, which can be something else